test/e2e/deployment.go

```diff
@@ -555,40 +559,117 @@ func (d *Deployment) DeleteResourcesForLocalContour() error {
 // Starts local contour, applying arguments and marshaling config into config
 // file. Returns running Contour command and config file so we can clean them
 // up.
-func (d *Deployment) StartLocalContour(config *config.Parameters, additionalArgs ...string) (*gexec.Session, string, error) {
-	configFile, err := ioutil.TempFile("", "contour-config-*.yaml")
-	if err != nil {
-		return nil, "", err
-	}
-	defer configFile.Close()
+func (d *Deployment) StartLocalContour(config *config.Parameters, contourConfiguration *contour_api_v1alpha1.ContourConfiguration, additionalArgs ...string) (*gexec.Session, string, error) {
+
+	var content []byte
+	var configReferenceName string
+	var contourServeArgs []string
+	var err error
+
+	// Look for the ENV variable to tell if this test run should use
+	// the ContourConfiguration file or the ContourConfiguration CRD.
+	if _, useContourConfiguration := os.LookupEnv("USE_CONTOUR_CONFIGURATION_CRD"); useContourConfiguration {
```

Comment:
If you change the GH Actions matrix, probably will end up being easier to check for a value of `true` here rather than just existence of the var.

Reply:
yup good point